projects/m2/012-shuffling-a-deck-of-cards/js/index.js

```diff
@@ -0,0 +1,27 @@
+function createDesk(){
+    // 52 card: values 2 through 9 and characters T, J, Q, K, A. Suit card: s, h, d, c
+    const values = ['2', '3', '4', '5', '6', '7', '8', '9', 'T', 'J', 'Q', 'K', 'A'];
+    const suitCard = ['s', 'h', 'd', 'c']
+    const card = []
+    // Use loop for creating list of cards
+    for(let suit of suitCard){
+        for(let item of values){
+        card.push(item + suit)
+        }
+    }
+    return card
+}
+
+// function shuffle order random the desk
+function shuffle(){
+    const desk = createDesk();
+    const newDesk = []
+    while(newDesk.length <= 52){
+        let randomDesk = Math.floor(Math.random() * desk.length)
+        newDesk.push(desk[randomDesk])
```

Comment:
I think you can **optimize** the **code**:
1) every time you **draw a card** from the **sorted desk**, it is also **eliminated**
2) as soon as the **desk** is **empty** and **newDesk** has **52 cards**, exit the while loop
As you have solved **currently**, it can happen to draw **double cards**(example **Ac**):
<img width="273" alt="image" src="https://user-images.githubusercontent.com/74595044/236618532-a77df521-183f-4af6-aa5a-441c5c7f1036.png">